test/e2e/set/scyllacluster/scyllamanager.go

```diff
@@ -267,82 +270,470 @@ var _ = g.Describe("Scylla Manager integration", func() {
+		o.Expect(sourceSC.Spec.Backups[0].Retention).To(o.Equal(int64(1)))
+
+		framework.By("Waiting for source ScyllaCluster to sync backup task update with Scylla Manager")
+		backupTaskUpdatedCond := func(cluster *scyllav1.ScyllaCluster) (bool, error) {
+			for _, b := range cluster.Status.Backups {
+				if b.Name == sourceSC.Spec.Backups[0].Name {
+					if b.ID == nil || len(*b.ID) == 0 {
+						return false, fmt.Errorf("got unexpected empty task ID in status")
+					}
+
+					if b.Error != nil {
+						return false, fmt.Errorf(*b.Error)
+					}
+
+					return b.Retention != nil && *b.Retention == int64(1), nil
+				}
+			}
+
+			return false, nil
+		}
+
+		waitCtx4, waitCtx4Cancel := utils.ContextForManagerSync(ctx, sourceSC)
+		defer waitCtx4Cancel()
+		sourceSC, err = controllerhelpers.WaitForScyllaClusterState(waitCtx4, f.ScyllaClient().ScyllaV1().ScyllaClusters(sourceSC.Namespace), sourceSC.Name, controllerhelpers.WaitForStateOptions{}, backupTaskUpdatedCond)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Expect(sourceSC.Status.Backups).To(o.HaveLen(1))
+		o.Expect(sourceSC.Status.Backups[0].Name).To(o.Equal(sourceSC.Spec.Backups[0].Name))
+		o.Expect(sourceSC.Status.Backups[0].Location).To(o.Equal(sourceSC.Spec.Backups[0].Location))
+		o.Expect(sourceSC.Status.Backups[0].Retention).NotTo(o.BeNil())
+		o.Expect(*sourceSC.Status.Backups[0].Retention).To(o.Equal(sourceSC.Spec.Backups[0].Retention))
+		o.Expect(sourceSC.Status.Repairs).To(o.BeEmpty())
+
+		framework.By("Verifying that updated backups task properties were synchronized")
+		tasks, err = managerClient.ListTasks(ctx, *sourceSC.Status.ManagerID, "backup", false, "", "")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(tasks.TaskListItemSlice).To(o.HaveLen(1))
+		backupTask = tasks.TaskListItemSlice[0]
+		o.Expect(backupTask.Name).To(o.Equal(sourceSC.Status.Backups[0].Name))
+		o.Expect(backupTask.ID).To(o.Equal(*sourceSC.Status.Backups[0].ID))
+		o.Expect(backupTask.Properties.(map[string]interface{})["location"]).To(o.ConsistOf(sourceSC.Status.Backups[0].Location))
+		o.Expect(backupTask.Properties.(map[string]interface{})["retention"].(json.Number).Int64()).To(o.Equal(*sourceSC.Status.Backups[0].Retention))
+
+		// Sanity check to avoid panics in the polling func.
+		o.Expect(sourceSC.Status.ManagerID).NotTo(o.BeNil())
+
+		var backupProgress managerclient.BackupProgress
+		framework.By("Waiting for backup to finish")
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 10*time.Minute, true, func(context.Context) (done bool, err error) {
+			backupProgress, err = managerClient.BackupProgress(ctx, *sourceSC.Status.ManagerID, backupTask.ID, "latest")
+			if err != nil {
+				return false, err
+			}
+
+			return backupProgress.Run.Status == managerclient.TaskStatusDone, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(backupProgress.Progress.SnapshotTag).NotTo(o.BeEmpty())
+
+		framework.By("Updating the backup task with invalid properties")
```

Comment:
Isn't this covered in https://github.com/scylladb/scylla-operator/pull/1908? This test should focus on backup + restore, not everything around.
As this one verifies also that existing task wasn't updated, I guess we could remove test added in 1908 as we have coverage for error propagation in both.

Reply:
The other test covers new invalid tasks, this covers invalid updates to tasks already in manager state and that they retain that state both in manager and our status. They don't overlap imo and this test didn't catch a bug I introduced when developing the fix, the other one did.

Reply:
They could be merged into one by adding an additional invalid task, or starting with invalid task creation, for both the manager/repair tests, but imo both are big enough as they are. Let me know what you think though.

Reply:
I was surprised when I was reading this test logic. Test name doesn't mention anything about testing task error propagation, while test added in #1908 does. I would move task error propagation on updates to 1908 and concentrate on backup and restore in this one.

Reply:
The other one would test backup/restore as a side effect then, so imo any way we do it, it kind of ends up with the same problem. Moving all error testing to one test case does seem marginally better though, as it makes this one more readable. I'll move this to #1908

Reply:
Moved to #1908